WireDomain/Sources/WireDomain/Repositories/TeamRepository.swift

```diff
@@ -72,6 +83,29 @@ final class TeamRepository: TeamRepositoryProtocol {
         await storeTeamLocally(team)
     }
 
+    func deleteTeamMember(
+        time: Date,
+        userID: UUID,
+        teamID: UUID
+    ) async throws {
+        let user = try await userRepository.fetchUser(with: userID)
+
+        try await context.perform { [context] in
+            guard let member = user.membership else {
+                throw TeamRepositoryError.userNotAMemberInTeam(user: userID, team: teamID)
+            }
+
+            if user.isSelfUser {
+                let notification = AccountDeletedNotification(context: context)
+                notification.post(in: context.notificationContext)
+            } else {
+                user.markAccountAsDeleted(at: time)
```

Comment:
I think we should move away from extension methods that perform logic from models. In my opinion, an object should be either plain old data, or something that performs logic with data.

I would suggest to move the logic hiding in the extension out, for instance, marking the user as deleted here, and then invoking a method on the conversation repo to remove that user from all conversations.

Reply:
it makes sense, even the user related storage logic could go the user repository, and inside the user repository calling the ConversationsRepository to delete the user from all conversations. 

e.g in the TeamRepo:

```
        try await context.perform { [context] in
            guard let member = user.membership else {
                throw TeamRepositoryError.userNotAMemberInTeam(user: userID, team: teamID)
            }
            
            userRepository.deleteUserAccount(user: user, leaveDate: time)

            context.delete(member)
        }

```

in the UserRepo:

```
    public func deleteUserAccount(user: ZMUser, leaveDate: Date) async {
        await context.perform { [self] in
            if user.isSelfUser {
                let notification = AccountDeletedNotification(context: context)
                notification.post(in: context.notificationContext)
            } else {
                user.isAccountDeleted = true
                conversationRepository.removeAllConversations(for: user)
            }
        }
    }

```

Reply:
What do you think @johnxnguyen  ?

Reply:
I've implemented it based on what was mentioned above, we have a new method in UserRepo to delete user account and another in ConversationRepo to remove the user from all conversations, so storage operations are split up between the dedicated repos. Merging this into the top branch https://github.com/wireapp/wire-ios/pull/1953.